### PR TITLE
refactor: minor changes to ElasticSearchRepository after comparing it with tasklist's version

### PR DESCRIPTION
**Pull request by entangled90**
## Description

First minor changes to ElasticSearchRepository after comparing it with tasklist implementation.

Prio{1,2,3,4}Backup interfaces now extend a common interface BackupPriority, which extends IndexRepository.
Thanks to this change, a small refactor in BackupServiceImpl eliminates all unchecked casts.

## Related issues

relates #24869

### CODE REVIEW

File: operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/BackupRestoreTest.java [hunk 1]
```diff
@@ -145,7 +145,7 @@ private void startAllApps() throws IOException {
     createSnapshotRepository(testContext);
 
     String zeebeVersion = ZeebeClient.class.getPackage().getImplementationVersion();
-    if (zeebeVersion.toLowerCase().contains("snapshot")) {
+    if (zeebeVersion == null || zeebeVersion.toLowerCase().contains("snapshot")) {
```
Comment (entangled90) [inline operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/BackupRestoreTest.java:148]: useful when running these tests from the IDE

File: webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java [hunk 2]
```diff
@@ -305,20 +307,22 @@ public List<SnapshotInfo> findSnapshots(final String repositoryName, final Long
...
         // no snapshot with given backupID exists
+        // Check Tasklist returns NotFoundApiException (Similar)
         throw new ResourceNotFoundException(
             String.format("No backup with id [%s] found.", backupId));
       }
       if (isRepositoryMissingException(e)) {
         final String reason =
             String.format("No repository with name [%s] could be found.", repositoryName);
-        throw new BackupRepositoryConnectionException(reason);
+        throw new GenericBackupException(reason);
```
Comment (entangled90) [inline webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java:317]: This exception type was raised incorrectly, same below

File: webapps-backup/src/main/java/io/camunda/webapps/backup/BackupServiceImpl.java [hunk 3]
```diff
@@ -133,53 +137,45 @@ void scheduleNextSnapshot() {
...
+        .toArray(String[]::new);
+  }
+
   // TODO Keep in mind this bug: https://github.com/camunda/camunda/issues/20458
   private String getCurrentVersion() {
     return backupProps.version().toLowerCase();
   }
+
+  private String getFullQualifiedName(final BackupPriority index) {
+    if (index != null) {
```
Comment (entangled90) [inline webapps-backup/src/main/java/io/camunda/webapps/backup/BackupServiceImpl.java:175]: unchecked casts removed here

File: operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/BackupRestoreTest.java (hunk 1 above)
Comment (npepinpe) [reply operate/qa/backup-restore-tests/src/test/java/io/camunda/operate/qa/backup/BackupRestoreTest.java:148]: :wrench: Add a comment explaining this

File: webapps-backup/src/main/java/io/camunda/webapps/backup/BackupServiceImpl.java [hunk 4]
```diff
@@ -93,18 +96,19 @@ public TakeBackupResponseDto takeBackup(final TakeBackupRequestDto request) {
...
-    return repository.getBackups(getRepositoryName());
+    return repository.getBackups(backupProps.repositoryName());
   }
 
   TakeBackupResponseDto scheduleSnapshots(final TakeBackupRequestDto request) {
-    final String repositoryName = getRepositoryName();
+    final String repositoryName = backupProps.repositoryName();
     final int count = getIndexPatternsOrdered().length;
     final List<String> snapshotNames = new ArrayList<>();
+    // FIXME potential issues
```
Comment (npepinpe) [inline webapps-backup/src/main/java/io/camunda/webapps/backup/BackupServiceImpl.java:111]: :question: What's this here for?

File: webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java [hunk 5]
```diff
@@ -305,20 +307,22 @@ public List<SnapshotInfo> findSnapshots(final String repositoryName, final Long
     } catch (final Exception e) {
       if (isSnapshotMissingException(e)) {
         // no snapshot with given backupID exists
+        // Check Tasklist returns NotFoundApiException (Similar)
```
Comment (npepinpe) [inline webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java:310]: :question: Is this a TODO?

File: webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java [hunk 6]
```diff
@@ -361,24 +365,23 @@ public boolean isSnapshotFinishedWithinTimeout(
...
       LOGGER.warn(
-          String.format(
-              "Snapshot state is %s for snapshot %s",
-              snapshotInfo.state(), snapshotInfo.snapshotId()));
+          "Snapshot state is {} for snapshot {}", snapshotInfo.state(), snapshotInfo.snapshotId());
       return false;
     }
   }
 
+  // Check: done with differences see ChecK: BELOW
```
Comment (npepinpe) [inline webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java:384]: :question: Are all these checks todos? :smile:

File: webapps-schema/src/main/java/io/camunda/webapps/schema/descriptors/backup/BackupPriority.java
Comment (npepinpe) [inline webapps-schema/src/main/java/io/camunda/webapps/schema/descriptors/backup/BackupPriority.java:1]: :question: Confused by Git here. Why does it think this is `BackupRepositoryPropsImpl` being moved to `BackupPriority`? :thinking:

**Review by npepinpe — APPROVED** [review]
Looks good generally, some questions, no blockers :+1:

File: webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java (hunk 5 above)
Comment (entangled90) [reply webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java:310]: Just a note for me when I will unify the controllers, to double check the exceptions as they are not checked by the compiler, I'll remove it when it's merged

File: webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java (hunk 6 above)
Comment (entangled90) [reply webapps-backup/src/main/java/io/camunda/webapps/backup/repository/elasticsearch/ElasticsearchBackupRepository.java:384]: They are mostly marks for the few differences between tasklist & operate implementation. I'll remove them when removing the impl in tasklist :+1:

File: webapps-schema/src/main/java/io/camunda/webapps/schema/descriptors/backup/BackupPriority.java
Comment (entangled90) [reply webapps-schema/src/main/java/io/camunda/webapps/schema/descriptors/backup/BackupPriority.java:1]: No idea, I deleted BackupRepositoryImpl and created BackupPriority from scratch :man_shrugging:

**Comment by npepinpe** [comment]
Just an FYI about the `Check` stuff, most IDEs have integrations to easily pick up `TODO` comments, so they're easy to spot/find, and we're less likely to forget them. Ultimately, if you handle all your comments that it doesn't matter, but if you're anything like me and likely to forget, `TODO` is a little easier :smile:

**Comment by entangled90** [comment]
They were actually `TODO` but the I reminded that if you commit via the IDE it checks that there are no TODO, so I didn't want to disruptor anybody's workflow

File: webapps-backup/src/main/java/io/camunda/webapps/backup/BackupServiceImpl.java (hunk 4 above)
Comment (entangled90) [reply webapps-backup/src/main/java/io/camunda/webapps/backup/BackupServiceImpl.java:111]: I'll remove it, I put it when we were wondering what to do about different versions in the backup name between tasklist & operate

**Pull request merged**